packages/client/src/build-id-types.ts

```diff
@@ -0,0 +1,137 @@
+import { temporal } from '@temporalio/proto';
+
+/**
+ * Operations that can be passed to {@link WorkflowClient.updateWorkerBuildIdCompatability}.
+ *
+ * @experimental
+ */
+export type BuildIdOperation =
+  | NewIdInNewDefaultSet
+  | NewCompatibleVersion
+  | PromoteSetByBuildId
+  | PromoteBuildIdWithinSet
+  | MergeSets;
+
+/**
+ * This operation adds a new Build Id into a new set, which will be used as the default set for
+ * the queue. This means all new workflows will start on this Build Id.
+ *
+ * @experimental
+ */
+interface NewIdInNewDefaultSet {
+  operation: 'NEW_ID_IN_NEW_DEFAULT_SET';
+  buildId: string;
+}
+
+/**
+ * This operation adds a new Build Id into an existing compatible set. The newly added ID becomes
+ * the default for that compatible set, and thus new workflow tasks for workflows which have been
+ * executing on workers in that set will now start on this new Build ID.
+ *
+ * @experimental
+ */
+interface NewCompatibleVersion {
+  operation: 'NEW_COMPATIBLE_VERSION';
+  // buildId The Build Id to add to an existing compatible set.
+  buildId: string;
+  // A Build Id which must already be defined on the task queue, and is used to
+  // find the compatible set to add the new ID to.
+  existingCompatibleBuildId: string;
+  // If set to true, the targeted set will also be promoted to become the
+  // overall default set for the queue.
+  makeSetDefault?: boolean;
+}
+
+/**
+ * This operation promotes a set of compatible Build IDs to become the current
+ * default set for the task queue. Any Build ID in the set may be used to
+ * target it.
+ *
+ * @experimental
+ */
+interface PromoteSetByBuildId {
+  operation: 'PROMOTE_SET_BY_BUILD_ID';
+  buildId: string;
+}
+
+/**
+ * This operation promotes a Build ID within an existing set to become the
+ * default ID for that set.
+ *
+ * @experimental
+ */
+interface PromoteBuildIdWithinSet {
+  operation: 'PROMOTE_BUILD_ID_WITHIN_SET';
+  buildId: string;
+}
+
+/**
+ * This operation merges two sets into one set, thus declaring all the Build Ids in both as
+ * compatible with one another. The default of the primary set is maintained as the merged set's
+ * overall default.
+ *
+ * @experimental
+ */
+interface MergeSets {
+  operation: 'MERGE_SETS';
+  // A Build Id which is used to find the primary set to be merged.
+  primaryBuildId: string;
+  // A Build Id which is used to find the secondary set to be merged.
+  secondaryBuildId: string;
+}
+
+/**
+ * Represents the sets of compatible Build ID versions associated with some
+ * Task Queue, as fetched by {@link WorkflowClient.getWorkerBuildIdCompatability}.
+ *
+ * @experimental
+ */
+export class WorkerBuildIdVersionSets {
```

Comment:
We don't return classes for simple objects like this all over the code base.
Instead define two interfaces:

```
export interface TaskQueueMetadata {
  versionSets: VersionSet[];
  defaultSet: string;
}

export interface VersionSet {
  buildIds: string[];
  defaultBuildId: string;
}
```

I regret that we didn't make the API return all of the task queue metadata `TaskQueueMetadata` as we're going to add more information to it.

For now you can name it `WorkerVersioningInfo`

Reply:
I'm not sure we should rename them as it's consistent with what I did in the other SDKs right now, but that's not the biggest deal. If you really prefer that we can rename